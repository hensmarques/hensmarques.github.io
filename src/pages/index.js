import React from "react"

import Layout from "../components/layout"
import About from "../components/about"
import Timeline from "../components/timeline"
import Portfolio from "../components/portfolio"
import Footer from "../components/footer"

import SEO from "../components/seo"

const IndexPage = () => (
    <Layout>
        <SEO title="Home" /> 
        <About></About>
        <Timeline></Timeline>
        <div className="mb-24"></div>
        <Portfolio></Portfolio>
        <Footer></Footer>
    </Layout>
)
    
export default IndexPage
    