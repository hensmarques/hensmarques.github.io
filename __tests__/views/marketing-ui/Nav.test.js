import React from 'react';
import { Nav } from '../../../src/views/marketing-ui/Nav';
import { authStatus } from '../../../src/reducers';

describe('<Nav/>', () => {
  describe('Non-authenticated user view', () => {
    const user = {
      authenticated: false
    };
    const onSignupClick = jest.fn();
    const onLoginClick = jest.fn();
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(
        <Nav user={user} />
      );
    });

    it('renders with signup and login links and matches snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Authenticated user view', () => {
    it('renders with a dashboard link instead of signup/login links', () => {
      const testAuthStatus = authStatus.success;
      const wrapper = shallow(<Nav authStatus={testAuthStatus} />);
      expect(wrapper.find('.signup').exists()).toBe(false);
      expect(wrapper.find('Link').exists()).toBe(true);
      const dashboardLink = wrapper.find('Link');
      expect(dashboardLink.props().to).toBe('/home');
      expect(dashboardLink.props().children).toBe('Dashboard');
    });
  });

  describe('Blacklist mode', () => {
    it('does not render signup, login or dashboard links', () => {
      const user = {
        authenticated: true
      };
      const wrapper = shallow(<Nav user={user} blacklistMode />);
      expect(wrapper.find('.signup').exists()).toBe(false);
      expect(wrapper.find('Link.login').exists()).toBe(false);
      expect(wrapper.find('Link').exists()).toBe(false);
    });
  });
});
