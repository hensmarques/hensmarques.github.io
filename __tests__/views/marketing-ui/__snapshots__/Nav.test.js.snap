// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav/> Non-authenticated user view renders with signup and login links and matches snapshot 1`] = `
<nav
  className="nav"
>
  <div
    className="nav-container"
  >
    <Scrollchor
      className="logo"
      to="#top"
    >
      <img
        className="logo-img"
        src="images/logos/company-logo-icon.png"
      />
    </Scrollchor>
    <ul
      className="nav-items"
    >
      <li>
        <Scrollchor
          animate={
            Object {
              "offset": -100,
            }
          }
          to="#features"
        >
          Features
        </Scrollchor>
      </li>
      <li>
        <Scrollchor
          animate={
            Object {
              "offset": -100,
            }
          }
          to="#about"
        >
          About Us
        </Scrollchor>
      </li>
      <li>
        <Scrollchor
          animate={
            Object {
              "offset": -100,
            }
          }
          to="#vision"
        >
          Our Vision
        </Scrollchor>
      </li>
      <li
        className="signup"
      >
        <Link
          replace={false}
          to="signup"
        >
          Sign Up
        </Link>
      </li>
      <li>
        <Link
          className="login-btn login"
          replace={false}
          to="login"
        >
          Login
        </Link>
      </li>
    </ul>
  </div>
</nav>
`;
